### PR TITLE
feat (gocardless): add gocardless API updates

### PR DESCRIPTION
**Pull request by lovrocolic**

### CODE REVIEW

File: docs/api/04_customers/create-update-customer.mdx [hunk 1]
```diff
@@ -273,6 +279,7 @@ If the customer already exists, the call will work as an update
 | -----------| -----| ----------- |
 | payment_provider | String &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | Payment provider used to bill the customer.<br/>Only accepted value: `stripe`, this field is required if you want to assign a `provider_customer_id`
 | provider_customer_id | String &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | Customer ID on the payment provider.<br/>If not provided, Lago can optionally create the provider customer
+| sync_with_provider | Boolean &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | Field that determines whether to create customer in payment provider or not
```
Comment (mathieu-lago) [inline docs/api/04_customers/create-update-customer.mdx:282]: I don't understand the difference between `sync_with_provider` and `sync`. Please can you explain?

File: docs/api/04_customers/customer-object.mdx [hunk 2]
```diff
@@ -70,8 +71,9 @@ It lets you create a customer, but also track usage and create invoices for the
 
 | Attributes | Description |
 | -----------| ----------- |
-| payment_provider &nbsp &nbsp <Type>String</Type> | Payment provider used to bill the customer. <details><summary>**Possible values**</summary><div>- `stripe`: Use stripe as a payment provider.<br></br>- null: No payment providers. Only webhooks will be send to process customer billing.<div></div></div></details> |
+| payment_provider &nbsp &nbsp <Type>String</Type> | Payment provider used to bill the customer. <details><summary>**Possible values**</summary><div>- `stripe`: Use stripe as a payment provider.<br></br>- `gocardless`: Use GoCardless as a payment provider.<br></br>- null: No payment providers. Only webhooks will be send to process customer billing.<div></div></div></details> |
```
Comment (mathieu-lago) [inline docs/api/04_customers/customer-object.mdx:74]: "Only webhooks will be sent to process billing."

File: docs/api/04_customers/create-update-customer.mdx (hunk 1 above)
Comment (lovrocolic) [reply docs/api/04_customers/create-update-customer.mdx:282]: `sync_with_provider` is a property that determines if customer will also be created in gocardless/stripe after creation in Lago. On the other hand, `sync` is only tech related field. It specifies if creation in payment provider will be handled in the sync/async way, but this field has no major effect on the 'product'

File: docs/api/04_customers/create-update-customer.mdx [hunk 3]
```diff
@@ -273,6 +279,7 @@ If the customer already exists, the call will work as an update
...
 | sync | Boolean &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | Set to true is you want to create the provider customer synchronously with the customer creation.<br/>Apply only when provider_customer_id is null and payment provider is configured to create customer.<br/>Default value: `false`
```
Comment (mathieu-lago) [inline docs/api/04_customers/create-update-customer.mdx:283]: Set to `true` if you want to create the customer record synchronously in the provider's system.<br/>Only applies when `provider_customer_id` is `null` and `sync_with_provider` is set to `true`.<br/>Default value: `false`

**Review by mathieu-lago — APPROVED** [review]

**Pull request merged**